src/Commands/RsyncCommand.php

```diff
@@ -64,6 +64,15 @@ public function rsyncCommand($src, $dest, array $rsyncOptions)
      */
     protected function rsync($site_env_id, $src, $dest, array $rsyncOptions)
     {
+        $maxAttempts = 5;
+
+        foreach ($rsyncOptions as $key => $option) {
+            if (strpos($option, '--max-attempts') === 0) {
+                $maxAttempts = (int) substr($option, strlen('--max-attempts') + 1);
+                unset($rsyncOptions[$key]);
+            }
```

Comment:
Maybe it's just me but it seems like we could do something more elegant with array instead of string logic.  This is a great effort though.

Reply:
This looks interesting: https://github.com/samejack/php-argv

Reply:
rsyncOptions is an array that contains all of the options that will be passed to the rsync command. Every option includes "--" because the command was called like this:

`terminus rsync -- --archive...`

So, with this logic we're searching for max-attempts option (created by us in this command) just to find out what's the value desired by the user and then remove it from the array so that this option doesn't get passed to rsync.

I think this is simple enough as to require a whole new external library. I'm happy to simplify it or add comments as long as it doesn't mean adding one more dependency to this project (doing that will mean that this is no longer compatible with terminus 2 so a new branch should be created, etc) :)

Reply:
Yeah, good point.  We need to keep downward compatibility because this still works on T1.

Reply:
On future plugins, that looks like a great library to leverage because argument parsing is something that is used a lot (justifying why adding the extra dependency is worthwhile).